README.md

```diff
@@ -324,6 +239,90 @@ def stream_openai_response(prompt: str):
         # Streaming metrics are automatically captured
 ```
 
+## 🔗 LangChain Integration
+
+Noveum Trace provides seamless integration with LangChain and LangGraph applications through a simple callback handler.
+
+### Quick Setup
+
+```python
+from noveum_trace.integrations import NoveumTraceCallbackHandler
+from langchain_openai import ChatOpenAI
+
+# Initialize Noveum Trace
+import noveum_trace
+noveum_trace.init(project="my-langchain-app", api_key="your-api-key")
+
+# Create callback handler
+handler = NoveumTraceCallbackHandler()
+
+# Add to your LangChain components
+llm = ChatOpenAI(callbacks=[handler])
+response = llm.invoke("What is the capital of France?")
+```
+
+### Advanced Features
+
+**Manual Trace Control**: For complex workflows, manually control trace lifecycle:
+
+```python
+# Start trace manually
+handler.start_trace("my-workflow")
+
+# Your LangChain operations
+llm = ChatOpenAI(callbacks=[handler])
+chain = LLMChain(llm=llm, prompt=prompt, callbacks=[handler])
+
+# End trace manually
+handler.end_trace()
+```
+
```

Comment:
Add docs for context manager here, we can remove advanced features from here from README and add this somewher else. We need to keep readme a bit simpler and quick to read.